server/src/remote.rs

```diff
@@ -296,19 +304,236 @@ impl FlightService for FlightServiceHandler {
         _request: Request<Empty>,
     ) -> Result<Response<Self::ListActionsStream>, Status> {
-        Err(Status::unimplemented("Not implemented."))
+        let create_table_action = ActionType {
+            r#type: "CreateTable".to_owned(),
+            description: "Given a table name and a schema, create a table and add it to the \
+            catalog.".to_owned(),
+        };
+
+        let create_model_table_action = ActionType {
+            r#type: "CreateModelTable".to_owned(),
+            description: "Given a table name, a schema, a list of tag column indices, and the \
+            timestamp column index, create a model table and add it to the catalog.".to_owned()
+        };
+
+        let output = stream::iter(vec![Ok(create_table_action), Ok(create_model_table_action)]);
+        Ok(Response::new(Box::pin(output)))
+    }
+}
+
+/// Given an array of bytes, extract a slice that contains the first argument. It is assumed that the
+/// length of the argument is in the first two bytes. A tuple with the argument bytes and an offset
+/// specifying where the next argument starts, is returned.
+fn extract_argument_bytes(data: &[u8]) -> (&[u8], usize) {
+    let size_bytes: [u8; 2] = data[..2].try_into().expect("Slice with incorrect length.");
+    let size = u16::from_be_bytes(size_bytes) as usize;
+
+    let argument_bytes = &data[2..(size + 2)];
+
+    (argument_bytes, size + 2)
+}
+
+/// Create a normal table and add it to the catalog. If the table already exists or if the Apache
+/// Parquet file cannot be created, return [`Status`] error.
+fn create_table(
+    mut catalog: RwLockWriteGuard<Catalog>,
+    table_name: String,
+    schema: Schema,
+) -> Result<(), Status> {
+    // If the table already exists, return an error.
+    let mut existing_tables = catalog.table_metadata.iter();
+    if existing_tables.any(|table| table.name == table_name) {
+        let message = format!("Table with name '{}' already exists.", table_name);
+        return Err(Status::already_exists(message));
     }
+
+    let file_name = format!("{}.parquet", table_name);
+    let file_path = catalog.data_folder_path.join(file_name);
+
+    // Create an empty Apache Parquet file to save the schema.
+    let empty_batch = RecordBatch::new_empty(Arc::new(schema));
+    StorageEngine::write_batch_to_apache_parquet_file(empty_batch, file_path.as_path())
+        .map_err(|error| Status::invalid_argument(error.to_string()))?;
+
+    // Save the table metadata in the catalog.
+    let path_str = file_path.to_str().unwrap().to_string();
+    let new_table_metadata = TableMetadata {
+        name: table_name.clone(),
+        path: path_str,
+    };
+
+    info!("Created table '{}'.", table_name);
+    catalog.table_metadata.push(new_table_metadata);
+
+    Ok(())
+}
+
+/// Create a model table, add it to the metadata database tables, and add it to the catalog.
+/// If the table already exists or if the model table cannot be created, return [`Status`] error.
```

Comment:
A check that ensures model tables never have more than 1024 columns should probably be added.

Reply:
Added a check and a test for the check.